src/Microsoft.VisualStudio.ProjectSystem.Managed/Tree/ProjectImportsSubTreeProvider.cs

```diff
@@ -0,0 +1,288 @@
+                                    }
+                                }
+
+                                // Process additions
+                                foreach (IProjectImportSnapshot import in imports)
+                                {
+                                    if (!knownImportPaths.Contains(import.ProjectPath))
+                                    {
+                                        IProjectTree newChild = NewTree(
+                                            Path.GetFileName(import.ProjectPath),
+                                            filePath: import.ProjectPath,
+                                            flags: ProjectImportFlags,
+                                            icon: s_nodeIcon);
+
+                                        // Recur down the tree
+                                        newChild = SyncNode(import.Imports, newChild);
+
+                                        tree = tree.Add(newChild).Parent!;
+                                    }
+                                }
+
+                                return tree;
+                            }
+                        }
+
+                        static IDisposable TrySuppressExecutionContextFlow()
+                        {
+                            return ExecutionContext.IsFlowSuppressed()
+                                ? EmptyDisposable.Instance
+                                : ExecutionContext.SuppressFlow();
+                        }
+                    }
+
+                    void TearDownTree()
+                    {
+                        Assumes.NotNull(_subscription);
+                        _subscription.Dispose();
+                        _subscription = null;
+
+                        SubmitTreeUpdateAsync(
+                            (currentTree, a, c) =>
+                            {
+                                IProjectTree tree = currentTree.Value.Tree;
+
+                                // Set invisible
+                                tree = tree.SetVisible(false);
+
+                                // Remove all children
+                                while (tree.Children.Count != 0)
+                                {
+                                    tree = tree.Children[0].Remove();
+                                }
+
+                                return Task.FromResult(new TreeUpdateResult(tree));
+                            });
+                    }
+                }
+            }
+        }
+
+        // CPS's physical tree provider currently only uses the IShowAllFilesProjectTreeProvider.ShowAllFiles property
+        // for graft root providers. These other interface methods have safe no-op implementations in case that changes
+        // one day.
+
+        bool IShowAllFilesProjectTreeProvider.CanIncludeItems(IImmutableSet<IProjectTree> nodes)   => true;
+        bool IShowAllFilesProjectTreeProvider.CanExcludeItems(IImmutableSet<IProjectTree> nodes)   => true;
```

Comment:
These two sound like they control menu items ("Exclude From Project"/"Include In Project") so `false` might be a safer option.

Reply:
These are specific to "Show All Files".